HACKING.md

```diff
@@ -39,3 +39,46 @@ To run the unit tests:
 To run the integration tests:
 
     ./integration-test.sh
+
+Importing Listenbrainz Listens to HDFS
+--------------------------------------
+
+To generate the stats and recommendations on your local machine you will need to import the listens into HDFS. It is recommended to use the ``listenbrainz-incremental-dumps`` for your local environment. Follow the given steps to import the listens.
+
+### Getting  name of the dump file
+
+Visit the [Listenbrainz FTP Server](http://ftp.musicbrainz.org/pub/musicbrainz/listenbrainz/) website and note down the name of the dump directory and the corresponding .tar.xz file you would like to import.
+
+### Update config.py
+
+Open the ``config.py`` file present in the ``listenbrainz_spark`` directory and look for the following section.
+
+    FTP_LISTENS_DIR = '/pub/musicbrainz/listenbrainz/fullexport/'
+
+Update the FTP listens directory to import the incremental dumps as given.
+
+    FTP_LISTENS_DIR = '/pub/musicbrainz/listenbrainz/incremental/'
+
+Now look for the given section in the same file.
+
+    TEMP_LISTENS_DIR = 'listenbrainz-dump-88-20190930-000002-full/'
+    TEMP_LISTENS_DUMP = 'listenbrainz-listens-dump-88-20190930-000002-spark-full.tar.xz'
+
+Update these strings with the directory name and the .tar.xz file you would like to import as temporary listens. e.g. To import the 	``listenbrainz-listens-dump-125-20200203-000002-incremental`` refer the below-given code.
+
+    TEMP_LISTENS_DIR = 'listenbrainz-dump-125-20200203-000002-incremental/'
+    TEMP_LISTENS_DUMP = 'listenbrainz-listens-dump-125-20200203-000002-incremental.tar.xz'
+
+### Import listens
+
+Next open the ``bash`` terminal for  ``listenbrainz_playground_1`` container by executing the following command
+
+    docker exec -it listenbrainzspark_playground_1 bash
```

Comment:
we should use `docker-compose exec` here, it should be possible to use develop.sh to call it.

Reply:
@alastair I tried running `docker-compose exec -it listenbrainzspark_playground_1 bash`. but this doesn't work. Please let me know where I am getting wrong?

Reply:
What is the error? Also, you did not specify the docker-compose file. Take a look at `develop.sh`, it might help.

Reply:
@alastair @vansika 
The equivalent command is 
`docker-compose -f docker/docker-compose.spark.yml -p listenbrainzspark exec playground bash` or 
`docker-compose -f docker/docker-compose.spark.yml -p listenbrainzspark run --rm playground bash`

It can also be run using
`./develop.sh spark exec playground bash`. or 
`./develop.sh spark run --rm playground bash`

I guess `./develop.sh spark run --rm playground bash` is better option since currently develop.sh uses `run` for most cases. Also for `exec -it`, although docker-compose automatically allocates TTY so `-t` is not needed, I couldn't get alternative for `-i` which keeps STDIN open even if not attached.

Let me know which command should I update in the documentation.

Reply:
the develop.sh spark commands look good. You're right that docker-compose doesn't need to specify -i or -t. I don't know why docker exec was used in this case, you'll have to verify if it was for a specific reason, in which case we should continue to use it, or if run will do the same thing.

Reply:
`exec` requires a running container and `run` creates a temporary container. It's good to use `exec` in this case. I will update it using `exec` command.